palvelutarjotin/oidc.py

```diff
@@ -19,7 +22,30 @@ def validate_claims(self, id_token):
         super().validate_claims(id_token)
 
 
-class GraphQLApiTokenAuthentication(FixTunnistamoMixin, ApiTokenAuthentication):
+class FixLastLoginMixin:
+    def update_last_login_after_interval(self, user):
+        """
+        Update the user object last_login field if enough time has passed.
+        The setting UPDATE_LAST_LOGIN_INTERVAL determines whether or not
+        enough time has passed for update.
+        We don't want to update on every request,
+        because then there would be too much writes when there are lot of users.
+        """
+        minutes_since_last_update = None
+        if user.last_login:
+            minutes_since_last_update = (
+                timezone.now() - user.last_login
+            ).total_seconds() / 60
+        if (
+            minutes_since_last_update is None
+            or minutes_since_last_update > settings.UPDATE_LAST_LOGIN["UPDATE_INTERVAL"]
+        ):
+            update_last_login(None, user)
```

Comment:
One has to know or look up the function signature to know what the first parameter `None` means here. Using keyword arguments here would add clarity IMHO and could be argued in favor of with Zen of Python's "Explicit is better than implicit":
```python
update_last_login(sender=None, user=user)
```
What do you think?

Reply:
Changed